WalletWasabi/Crypto/Random/SecureRandom.cs

```diff
@@ -0,0 +1,47 @@
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.WabiSabi.Crypto
+{
+	public static class SecureRandom
+	{
+		public static Scalar GetScalarNonZero()
+		{
+			using var randomGenerator = RandomNumberGenerator.Create();
```

Comment:
Consider using the NBitcoin RandomUtils with the following settings:

```
	public class RandomWithSeed : IRandom
	{
		private Random _random;

		public RandomWithSeed(int seed)
		{
			this._random =  new Random(seed);
		}

		public void GetBytes(byte[] output)
		{
			_random.NextBytes(output);
		}

		public void GetBytes(Span<byte> output)
		{
			_random.NextBytes(output);
		}

		public byte[] GetNextBytes(int size)
		{
			var buffer = new byte[size];
			GetBytes(buffer);
			return buffer;
		}
	}

	RandomUtils.Random = new RandomWithSeed(123456);
	RandomUtils.UseAdditionalEntropy = false;
```

Because in thise way if a method fails for some reason we will be able to reproduce it and fix it. Otherwise a test can fail today and never fail again in a year. Also, NBitcoin internally uses `RandomUtils` for absolutely everything so, it can generate the same keys, transactions, blocks, etc again and again in case we use it consistently.

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/3869